### DISCUSSION TITLE
Confirm transaction details in Identity Approval screen

### DISCUSSION THREAD

**Discussion by tijno** [deso-protocol/dips#97] [Medium CIPs]
@AdityaChaudhary0005 posted this valid risk on Bitclout

https://bitclout.com/posts/6c70ba45b334334ade3077766dea3e2342e680edc0bb238eecff77d45fc9294b

## Problem

> But some evil hands can design their frontend where doesn't matter where you want to send CLOUT, all transactions will be made to a specific XYZ account. It will seem that you are doing it all right but behind the scenes you won't even realised that you have been fooled.

To somewhat address this it would make sense for Identity to show a summary of the transaction you are approving.

Currently it only shows the type of transaction but not the details.

![20210726-p0S109Od](https://user-images.githubusercontent.com/69529928/126993720-4122edd8-4b2f-4f4b-a63e-4256ebba1ed7.png)

## Simple mockup

![20210726-vZHBolsj](https://user-images.githubusercontent.com/69529928/126993775-5b0aa512-b715-4791-b688-9669526cab9e.png)

## Implementation

Add details to this div:
https://github.com/bitclout/identity/blob/main/src/app/approve/approve.component.html#L7

Using the `description` variables in this Switch:

https://github.com/bitclout/identity/blob/main/src/app/approve/approve.component.ts#L80

**Comment by ItsAditya-xyz** [comment]
+1 to this. This brings more transparency during a transaction. Indeed a good solution to the problem I mentioned in my post.

**Comment by FreeTrade** [comment]
Agree.

One wrinkle - I'm wondering how Identity will resolve the BitClout address to a username.

**Reply by ItsAditya-xyz** [reply]
APIs ?

**Reply by tijno** [reply]
good point as its not in the transaction - unless nodes send it to identity. API could also solve but then identity becomes dependant on that api.

it can be solved of course - just needs to be done in the right way
